tests/test_iri.py

```diff
@@ -0,0 +1,119 @@
+)
+from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+
+
+@pytest.fixture()
+def mock_download(mocker, tmp_path):
+    """
+    Call download with mocked _download.
+
+    `forecast_type` is the type of forecast to
+    test, can be either 'tercile' or 'dominant'.
+    """
+    download_mock = mocker.patch(
+        "aatoolbox.datasources.iri."
+        "iri_seasonal_forecast._IriForecast._download"
+    )
+
+    def _mock_download(forecast_type):
+        iso3 = "abc"
+        geo_bounding_box = GeoBoundingBox(north=6, south=3.2, east=-2, west=3)
+        if forecast_type == "tercile":
+            iri_class = IriForecastTercile(
+                iso3=iso3, geo_bounding_box=geo_bounding_box
+            )
+        elif forecast_type == "dominant":
+            iri_class = IriForecastDominant(
+                iso3=iso3, geo_bounding_box=geo_bounding_box
+            )
+        iri_class._raw_base_dir = tmp_path
+        iri_class.download(iri_auth=None)
+        _, kwargs_download = download_mock.call_args
+        url = kwargs_download["url"]
+        filepath = kwargs_download["filepath"]
+        return url, filepath, tmp_path
+
+    return _mock_download
+
+
+def test_download_call_tercile(mock_download):
+    """Test download for tercile forecast."""
+    url, filepath, tmp_path = mock_download("tercile")
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+        ".NMME_Seasonal_Forecast/"
+        ".Precipitation_ELR/.prob/X/%283.0%29%28-2.0%29RANGEEDGES/"
+        "Y/%286.0%29%283.0%29RANGEEDGES/data.nc"
+    )
+
+    assert (
+        filepath
+        == tmp_path
+        / "abc_iri_forecast_seasonal_precipitation_tercile_Np6Sp3Em2Wp3.nc"
+    )
+
+
+def test_download_call_dominant(mock_download):
+    """Test download for tercile forecast."""
+    url, filepath, tmp_path = mock_download("dominant")
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+        ".NMME_Seasonal_Forecast/"
+        ".Precipitation_ELR/.dominant/X/%283.0%29%28-2.0%29RANGEEDGES/"
+        "Y/%286.0%29%283.0%29RANGEEDGES/data.nc"
+    )
+
+    assert (
+        filepath == tmp_path / "abc_iri_forecast_seasonal_precipitation_"
+        "tercile_dominant_Np6Sp3Em2Wp3.nc"
+    )
+
+
+def test_process(tmp_path):
+    """Test process for IRI forecast."""
+    ds = xr.DataArray(
+        np.reshape(a=np.arange(16), newshape=(2, 2, 2, 2)),
+        coords=[
+            [1, 2],
+            [3, -2],
+            [97, 90],
+            [685.5, 686.5],
+        ],
+        dims=["L", "X", "Y", "F"],
+    ).to_dataset(name="prob")
+    ds["F"].attrs["calendar"] = "360"
+    ds["F"].attrs["units"] = "months since 1960-01-01"
+    iso3 = "abc"
+    geo_bounding_box = GeoBoundingBox(north=6, south=3.2, east=-2, west=3)
+    iri_class = IriForecastTercile(
+        iso3=iso3, geo_bounding_box=geo_bounding_box
+    )
+
+    iri_class._process(filepath=tmp_path / "test.nc", ds=ds, clobber=False)
```

Comment:
arghh, optimally I would like to test `process` instead of `_process`. I thought I could do so by mocking the loading of the dataset, i.e. `mocker.patch( "aatoolbox.datasources.iri.iri_seasonal_forecast.xr.load_dataset", return_value = ds )` 
But that is giving strange results.. @turnerm any idea how I could do this?

Reply:
Yeah exactly, you should mock the loading and pass it a fake dataset instead

Reply:
Yeah so this somehow wasn't working. I still don't fully understand why but when I was doing it as I described above, no errors showed up but the returned dataset after calling `.process()` was the original dataset and not the processed one. 

I now created an extra function in `iri_seasonal_forecast` named `load_raw`. In the test I then mock that function, and that works fine. 
I am not super happy with needing an extra function just for the purpose of mocking but couldn't come up with a better method. If you know a better way to do this, I am happy to hear!